beluga_ros/include/beluga_ros/ndt_ellipsoid.hpp

```diff
@@ -0,0 +1,146 @@
+  }
+  const auto rotation = Eigen::Quaterniond{rotationMatrix};
+
+  // Fill in the message
+  marker.type = beluga_ros::msg::Marker::SPHERE;
+  marker.action = beluga_ros::msg::Marker::ADD;
+
+  marker.pose.position.x = cell.mean[0];
+  marker.pose.position.y = cell.mean[1];
+  marker.pose.position.z = cell.mean[2];
+
+  marker.pose.orientation.x = rotation.x();
+  marker.pose.orientation.y = rotation.y();
+  marker.pose.orientation.z = rotation.z();
+  marker.pose.orientation.w = rotation.w();
+
+  float scaleConstant = 2.0f;
+  marker.scale.x = std::sqrt(scalevector.x()) * scaleConstant;
+  marker.scale.y = std::sqrt(scalevector.y()) * scaleConstant;
+  marker.scale.z = std::sqrt(scalevector.z()) * scaleConstant;
+
+  return true;
+}
+
+inline void use_cell_size(const auto& position, const auto& size, auto& marker){
+  marker.type = beluga_ros::msg::Marker::CUBE;
+  marker.action = beluga_ros::msg::Marker::ADD;
+
+  marker.pose.position.x = position[0];
+  marker.pose.position.y = position[1];
+  marker.pose.position.z = position[2];
+
+  marker.scale.x = size;
+  marker.scale.y = size;
+  marker.scale.z = size;
+}
+
+/// Assign an ellipsoid to each cell of a SparseValueGrid. A cube is used instead if the distribution of the
+/// cell is not suitable for the rotation matrix creation.
+/** 
+ * \param grid A SparseValueGrid that contains cells representing obstacles.
+ * \return A message with the ellipsoids or cubes.
+ * \param[out] message Markers message to be assigned.
+ * \tparam MapType Container that maps from Eigen::Vector<int, NDim> to the type of the cell. See [SparseValueGrid]
+ * (https://ekumen-os.github.io/beluga/packages/beluga/docs/_doxygen/generated/reference/html/classbeluga_1_1SparseValueGrid.html).
+ * \tparam NDim Dimension of the grid.
+ */
+template <typename MapType, int NDim>
+beluga_ros::msg::MarkerArray assign_obstacle_map(
+    const beluga::SparseValueGrid<MapType, NDim>& grid,
+    beluga_ros::msg::MarkerArray& message) {
+  
+  // Get data from the grid
+  auto& map = grid.data();
+
+  // Clean up the message
+  beluga_ros::msg::Marker marker;
+  marker.ns = "obstacles";
+  marker.action = beluga_ros::msg::Marker::DELETEALL;
+  message.markers.push_back(marker);
+
+  // Add the markers
+  Eigen::SelfAdjointEigenSolver<Eigen::Matrix<double, 3, 3>> eigenSolver;
+  for (auto [index, entry] : ranges::views::enumerate(map)) {
+    const auto& [cellCenter, cell] = entry;
+    beluga_ros::msg::Marker marker;
+    marker.header.frame_id = "map";
+    marker.id = index;
+    marker.ns = "obstacles";
+
+    // Try to create an ellipsoid with values of the cell
+    if(!use_mean_covariance(eigenSolver, cell, marker)){
+      // Create a cube based on the resolution of the grid
+      use_cell_size(cellCenter, grid.resolution(), marker);
+    }
+
+    float color[] = {0.0f, 1.0f, 0.0f, 1.0f};
+    marker.color.r = color[0];
+    marker.color.g = color[1];
+    marker.color.b = color[2];
+    marker.color.a = color[3];
```

Comment:
@Fernando-Sanz why not simply assign literals?

Reply:
It was a try to see the color code better. I'll change it to the previous way.